src/index.js

```diff
@@ -0,0 +1,46 @@
+import './style.css';
+import './logo.jpg';
+import {
+  fetchCategoryMeals, displayMeals, sendLike, updateLikesCount,
+} from './modules/render.js';
+import { updateCategoryCounts } from './modules/mealCounter.js';
+
+const headerLinks = document.querySelectorAll('header a');
+const mealListCont = document.getElementById('mealList');
+
+const renderCategory = async (category) => {
+  const categoryMeals = await fetchCategoryMeals(category);
+  displayMeals(categoryMeals);
+};
+
+const setupHeaderLinkListeners = () => {
+  headerLinks.forEach((link) => {
+    const { dataset: { category } } = link;
+    link.addEventListener('click', async (e) => {
+      e.preventDefault();
+      await renderCategory(category);
+    });
+  });
+};
+
+const initializeApp = async () => {
+  await updateCategoryCounts();
+  setupHeaderLinkListeners();
+  renderCategory('Seafood');
+};
+
+mealListCont.addEventListener('click', async (e) => {
+  if (e.target.classList.contains('likes-icon')) {
+    const mealTitleElement = e.target.closest('.meal').querySelector('h3');
+    const mealTitle = mealTitleElement.textContent;
+    const likesCountElement = e.target.nextElementSibling;
+    const currentLikes = parseInt(likesCountElement.textContent, 10);
+    const updatedLikes = currentLikes + 1;
+    const success = await sendLike(mealTitle);
+    if (success) {
+      updateLikesCount(likesCountElement, updatedLikes);
```

Comment:
- [ ] Great job working on this capstone project. As per project requirements, you are supposed to used the Involvement API to show the item likes which I don't think is the case at the moment. You are only using the Involvement API at the moment for the comments counter. Kindly make use of this API for item likes.
![image](https://github.com/tajemouti/js-capstone-project/assets/62849872/dfa1f123-a88f-4095-9a21-ebd54f4b3271)

Reply:
## Hi @t-yanick ##
Thank you for the fast and insightful review. I really appreciate it.
Actually, we used the Involvement API to render the likes to the meals cards, the displayMeals function in the render.js file. This function is called when rendering each meal on the page, and it involves fetching the number of likes for each meal from the Involvement API and updating the corresponding likes count element in the HTML.

- Here's the relevant part of the displayMeals function:

1. `const likes = await fetchLikes(meal.strMeal);`:

This line fetches the number of likes for the current meal using the fetchLikes function, passing the meal title (meal.strMeal) as an argument.

2. `updateLikesCount(likesCount, likes);`:
 
This line calls the updateLikesCount function, which takes the likes count element (likesCount) and the fetched likes value (likes) as arguments. The purpose of this call is to update the displayed likes count for the current meal in the HTML.

- Now, allow me to kindly walk you through this piece of code line by line:

mealListCont.addEventListener('click', async (e) => {
  if (e.target.classList.contains('likes-icon')) {
    const mealTitleElement = e.target.closest('.meal').querySelector('h3');
    const mealTitle = mealTitleElement.textContent;
    const likesCountElement = e.target.nextElementSibling;
    const currentLikes = parseInt(likesCountElement.textContent, 10);
    const updatedLikes = currentLikes + 1;
    const success = await sendLike(mealTitle);
    if (success) {
      updateLikesCount(likesCountElement, updatedLikes);
    }
  }
});

This piece of code is responsible for handling the user interaction when the "Like" button (heart icon) for a specific meal in the web application is clicked. Let's break down what each part of the code is doing:

1. `mealListCont.addEventListener('click', async (e) => { ... })`:
   - This adds a click event listener to the `mealListCont` element, which is the container for displaying meals on the webpage.
   - The listener is set up to react to click events on any element inside the `mealListCont` container.

2. `if (e.target.classList.contains('likes-icon')) { ... }`:
   - This conditional statement checks whether the clicked element has a class of `'likes-icon'`. It's used to specifically target the heart icon (like button) that was clicked.

3. `const mealTitleElement = e.target.closest('.meal').querySelector('h3');`:
   - This line finds the closest ancestor element with the class `'meal'` from the clicked element, then selects the `<h3>` element within that ancestor.
   - This `<h3>` element contains the title of the meal.

4. `const mealTitle = mealTitleElement.textContent;`:
   - This extracts the text content (title) of the selected meal's `<h3>` element.

5. `const likesCountElement = e.target.nextElementSibling;`:
   - This selects the next sibling element of the clicked element. In this case, it's the element that displays the number of likes for the corresponding meal.

6. `const currentLikes = parseInt(likesCountElement.textContent, 10);`:
   - This converts the text content of the element that displays the current number of likes into an integer. This allows us to perform numerical calculations.

7. `const updatedLikes = currentLikes + 1;`:
   - This calculates the updated number of likes after the user clicks the like button.

8. `const success = await sendLike(mealTitle);`:
   - This invokes the `sendLike` function with the `mealTitle` as an argument.
   - The `sendLike` function sends a request to the Involvement API to record the user's like for the specific meal.
   - The `await` keyword ensures that the function waits for the API request to complete and returns a success status.

9. `if (success) { ... }`:
   - If the `sendLike` function call was successful (meaning the API request was successful), this condition is met.

10. `updateLikesCount(likesCountElement, updatedLikes);`:
    - This calls the `updateLikesCount` function to update the displayed likes count on the webpage.
    - The function takes the `likesCountElement` (the DOM element displaying likes count) and the `updatedLikes` value as arguments.

In summary, this code listens for clicks on the "Like" button for a meal, then retrieves the meal title and current likes count. After that, it sends a like to the Involvement API, and if the API request is successful, it updates the displayed likes count on the webpage.

- Finally, here's a live [demo](https://drive.google.com/file/d/1EBDBThRFdnC1hiNZ_u6E1tUoU5vGxyeZ/view?usp=sharing) that shows clearly the interaction of the Likes functionality with the Involvement API.

Thank you for taking the time to go through all of this.
Cheers